### ISSUE TITLE
SyntaxError: Unexpected token: < (line 132).

### ISSUE THREAD

**Issue by mjoselli** [cryp-doh/cmc-google-apps#2]
Hello,

When I try to get the data with something like =CMC("ETH/USD")
I get this error: SyntaxError: Unexpected token: < (line 132).

**Comment by mtompkins** [comment]
+1

**Comment by bryanjjones** [comment]
I finally tracked down the source of this error.

**Comment by bryanjjones** [comment]
I finally tracked down the source of this error. 
TLDR: Coinmarketcap.com is blocking requests from google apps.  

Everybody using this code (or any google docs app like this, including the cryptofinance app) is requesting data from coinmarketcap from one or a small number of shared IPs (e.g. 35.187.133.76). Since the script is running server side, all the requests come from the google cloud servers, not from your local machine. Coinmarketcap.com uses a service called "Cloudflare" to block specific IP addresses, presumably from people who are spamming the API. Since so many people are trying to ping Coinmarketcap using google app scripts like this (and not everyone is being responsible in their use and number of requests), their service has blocked them all. It's not a universal block, it's a limit, so you might get lucky from time to time and get a request through. 
Unfortunately, I don't have a fix. I'd love to find a workaround. 
Below is the full html code that is returned instead of the json data that is expected by the script. You can see there are a bunch of "<" characters that are not what is expected, which explains the (uninformative) error "SyntaxError: Unexpected token: < (line 132)"

```
 <!DOCTYPE html>
<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en-US"> <![endif]-->
<!--[if IE 7]>    <html class="no-js ie7 oldie" lang="en-US"> <![endif]-->
<!--[if IE 8]>    <html class="no-js ie8 oldie" lang="en-US"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en-US"> <!--<![endif]-->
<head>
<title>Access denied | api.coinmarketcap.com used Cloudflare to restrict access</title>
<meta charset="UTF-8" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1" />
<meta name="robots" content="noindex, nofollow" />
<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1" />
<link rel="stylesheet" id="cf_styles-css" href="/cdn-cgi/styles/cf.errors.css" type="text/css" media="screen,projection" />
<!--[if lt IE 9]><link rel="stylesheet" id='cf_styles-ie-css' href="/cdn-cgi/styles/cf.errors.ie.css" type="text/css" media="screen,projection" /><![endif]-->
<style type="text/css">body{margin:0;padding:0}</style>


<!--[if gte IE 10]><!--><script type="text/javascript" src="/cdn-cgi/scripts/zepto.min.js"></script><!--<![endif]-->
<!--[if gte IE 10]><!--><script type="text/javascript" src="/cdn-cgi/scripts/cf.common.js"></script><!--<![endif]-->



</head>
<body>
  <div id="cf-wrapper">
    <div class="cf-alert cf-alert-error cf-cookie-error" id="cookie-alert" data-translate="enable_cookies">Please enable cookies.</div>
    <div id="cf-error-details" class="cf-error-details-wrapper">
      <div class="cf-wrapper cf-header cf-error-overview">
        <h1>
          <span class="cf-error-type" data-translate="error">Error</span>
          <span class="cf-error-code">1015</span>
          <small class="heading-ray-id">Ray ID: 48b3928eaa2f5849 &bull; 2018-12-18 18:08:02 UTC</small>
        </h1>
        <h2 class="cf-subheadline">You are being rate limited</h2>
      </div><!-- /.header -->

      <section></section><!-- spacer -->

      <div class="cf-section cf-wrapper">
        <div class="cf-columns two">
          <div class="cf-column">
            <h2 data-translate="what_happened">What happened?</h2>
            <p>The owner of this website (api.coinmarketcap.com) has banned you temporarily from accessing this website.</p>
          </div>

          
        </div>
      </div><!-- /.section -->

      <div class="cf-error-footer cf-wrapper">
  <p>
    <span class="cf-footer-item">Cloudflare Ray ID: <strong>48b3928eaa2f5849</strong></span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span>Your IP</span>: 35.187.133.76</span>
    <span class="cf-footer-separator">&bull;</span>
    <span class="cf-footer-item"><span>Performance &amp; security by</span> <a href="https://www.cloudflare.com/5xx-error-landing?utm_source=error_footer" id="brand_link" target="_blank">Cloudflare</a></span>
    
  </p>
</div><!-- /.error-footer -->


    </div><!-- /#cf-error-details -->
  </div><!-- /#cf-wrapper -->

  <script type="text/javascript">
  window._cf_translation = {};
  
  
</script>

</body>
</html>
```

**Comment by jonasrk** [comment]
Thanks all. This renders the code in this repo useless for the moment, doesn't it?
